plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -189,6 +209,43 @@ public void shouldIgnoreColumnsWhenBuildMultipleLineAnnotation() {
                 .hasFieldOrPropertyWithValue("endColumn", Optional.empty());
     }
 
+    /**
+     * Test that publishIssues honors the checks name provided by a withChecks context.
+     */
+    @Test
+    public void shouldHonorWithChecksContextPublishIssues() {
+        WorkflowJob project = createPipeline();
+        copySingleFileToWorkspace(project, NEW_CHECKSTYLE_REPORT);
+        project.setDefinition(asStage("withChecks('Custom Checks Name') {", createScanForIssuesStep(new CheckStyle()), PUBLISH_ISSUES_STEP, "}"));
+        buildSuccessfully(project);
+
+        List<ChecksDetails> publishedChecks = PUBLISHER_FACTORY.getPublishedChecks();
+
+        assertThat(publishedChecks.size()).isEqualTo(2);
+
+        publishedChecks.forEach(check -> assertThat(check.getName()).isPresent().get().isEqualTo("Custom Checks Name"));
+    }
+
+    /**
+     * Test that recordIssues honors the checks name provided by a withChecks context.
+     */
+    @Test
+    public void shouldHonorWithChecksContextRecordIssues() {
+        WorkflowJob project = createPipeline();
+        copySingleFileToWorkspace(project, NEW_CHECKSTYLE_REPORT);
+        project.setDefinition(asStage(String.format(""
+                + "withChecks('Custom Checks Name') {\n"
+                + "  recordIssues(tools: [%s(pattern: '%s')])\n"
+                + "}", new CheckStyle().getSymbolName(), NEW_CHECKSTYLE_REPORT)));
+        buildSuccessfully(project);
+
+        List<ChecksDetails> publishedChecks = PUBLISHER_FACTORY.getPublishedChecks();
+
+        assertThat(publishedChecks.size()).isEqualTo(2);
+
+        publishedChecks.forEach(check -> assertThat(check.getName()).isPresent().get().isEqualTo("Custom Checks Name"));
```

Comment:
This is not actually part of your PR: but since you added a simple way to verify the checks can you also add another test that does not wrap the code with `withChecks` and see that the original title is used?